src/tr1d1um/communication.go

```diff
@@ -50,8 +54,24 @@ func (tr1 *Tr1SendAndHandle) Send(ch *ConversionHandler, resp http.ResponseWrite
 	//todo: any more headers to be added here
 	requestToServer.Header.Set("Content-Type", wrp.JSON.ContentType())
 	requestToServer.Header.Set("Authorization", req.Header.Get("Authorization"))
-	respFromServer, err = tr1.timedClient.Do(requestToServer)
-	return
+
+	ctx, cancel := context.WithTimeout(context.Background(), tr1.timeout)
+	defer cancel()
+
+	requestWithContext := requestToServer.WithContext(ctx)
+	ready := make(chan bool)
```

Comment:
idiomatic go is to use `ready := make(chan struct{})`, then just `defer close(ready)` inside the goroutine.  the `select` below stays the same.